internal/services/mssql/mssql_virtual_machine_resource.go

```diff
@@ -285,7 +285,12 @@ func resourceMsSqlVirtualMachine() *pluginsdk.Resource {
 						},
 						"data_settings":    helper.StorageSettingSchema(),
 						"log_settings":     helper.StorageSettingSchema(),
-						"temp_db_settings": helper.StorageSettingSchema(),
+						"temp_db_settings": helper.SQLTempDBStorageSettingSchema(),
```

Comment:
To use the new properties in `SQLTempDBStorageSettingSchema`, functions `expandSqlVirtualMachineTempDbSettings` and `flattenSqlVirtualMachineTempDbSettings` need to be updated as well to include these, otherwise these would not be included in API request or set into state file.

Reply:
@myc2h6o Ok I think I saw what you meant, and I think I got it updated.

Reply:
Thanks for the update. It looks good to me